std/conv.d

```diff
@@ -5338,3 +5338,229 @@ unittest
+{
+    return hexStrImpl!(C)(hexData);
+}
+
+/// Ditto
+@property @safe pure
+auto hexString(C = char)(string hexData)
+if (isSomeChar!C)
+{
+    if (hexData.isHexLitteral)
+        return hexStrImpl!(C)(hexData);
+    else
+        assert(0, "Invalid input string format in " ~ __FUNCTION__);
+}
+
+///
+unittest
+{
+    // conversion at compile time
+    auto string1 = hexString!"304A314B";
+    assert(string1 == "0J1K");
+    // conversion at run time
+    auto arbitraryData = "30 4A 31 4B";
+    auto string2 = hexString(arbitraryData);
+    assert(string2 == "0J1K");
+}
+
+/**
+The $(D hexBytes) function is similar to the $(D hexString) one except
+that it returns the hexadecimal data as an array of integral.
+
+Params:
+hexData = the $(D string) to be converted.
+T = the integer type of an array element. By default it is set to $(D ubyte) 
+but all the integer type, excepted $(D ulong) and $(D long), are accepted. 
+ */
+@property @trusted nothrow pure
+auto hexBytes(string hexData, T = ubyte)()
+if (hexData.isHexLitteral && isIntegral!T && (T.sizeof <= 4))
+{
+    static if (T.sizeof == 1)
+        return cast(T[]) hexStrImpl!char(hexData);
+    else static if (T.sizeof == 2)
+        return cast(T[]) hexStrImpl!wchar(hexData);
+    else
+        return cast(T[]) hexStrImpl!dchar(hexData);
+}
+
+/// ditto
+@property @trusted pure
+auto hexBytes(T = ubyte)(string hexData)
+if (isIntegral!T && (T.sizeof <= 4))
+{
+    if (hexData.isHexLitteral)
+    {
+        static if (T.sizeof == 1)
+            return cast(T[]) hexStrImpl!char(hexData);
+        else static if (T.sizeof == 2)
+            return cast(T[]) hexStrImpl!wchar(hexData);
+        else
+            return cast(T[]) hexStrImpl!dchar(hexData);
+    }
+    else assert(0, "Invalid input string format in " ~ __FUNCTION__);
+}
+
+
+///
+unittest
+{
+    // conversion at compile time
+    auto array1 = hexBytes!"304A314B";
+    assert(array1 == [0x30, 0x4A, 0x31, 0x4B]);
+    // conversion at run time
+    auto arbitraryData = "30 4A 31 4B";
+    auto array2 = hexBytes(arbitraryData);
+    assert(array2 == [0x30, 0x4A, 0x31, 0x4B]);
+}
+
+/*
+    takes a hexadecimal string literal and returns its representation.
+    hexData is granted to be a valid string by the caller.
+    C is granted to be a valid char type by the caller.
+*/
+@safe nothrow pure  
+private auto hexStrImpl(C)(string hexData)
+{
+    import std.ascii;
+    immutable(C)[] result; 
+    ubyte chr, cnt;
+    foreach(c; hexData) 
+        if (c.isHexDigit)
+        {
+            if (cnt == 0)
+            {
+                chr = 0;
+                if (c.isAlpha) 
+                    chr += (c.toLower - 'a' + 10) << 4;
+                else    
+                    chr += (c - '0') << 4;
+            }
+            else
+            {
+                if (c.isAlpha) 
+                    chr += (c.toLower - 'a' + 10); 
+                else    
+                    chr += (c - '0');
+                result ~= chr;
```

Comment:
To avoid GC spam it's possible to set .length at the beginning based on the input string, define each char/wchar/dchar with [], and at the end set .length again according to the count of chars which have been added.

Any other suggestion ?